### PR TITLE
[feat](nereids)support refresh table command

### LINKED ISSUE #42578: [Enhancement] (nereids)implement RefreshTableCommand in nereids
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

RefreshTableCommand is supported in legacy planner but not in nereids(new doris fe planner). you need implement it in nereids.
The DorisParser.g4 has the syntax definition already(see bellow). And you can check RefreshTableStmt.java in legacy planner to see what the command does and migrate it to nereids planner
```unsupportedRefreshStatement : REFRESH TABLE name=multipartIdentifier```

### Solution

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vinlee19**
### What problem does this PR solve?

Issue Number: close https://github.com/apache/doris/issues/42578

Related PR: #xxx

Problem Summary:
support refresh table command in nereids.
### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run feut

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/refresh/RefreshTableCommand.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        super(PlanType.REFRESH_TABLE_COMMAND);
+        this.tableNameInfo = tableName;
+    }
+
+    @Override
+    public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        tableNameInfo.analyze(ctx);
+        checkRefreshTableAccess();
+        // refresh table execute logic
+        refreshTable(tableNameInfo.getCtl(), tableNameInfo.getDb(), tableNameInfo.getTbl(), false);
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/refresh/RefreshTableCommand.java:62]:
```suggestion
        Env.getCurrentEnv().getRefreshManager().refreshTable(tableNameInfo.getCtl(), tableNameInfo.getDb(), tableNameInfo.getTbl(), false);
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/refresh/RefreshTableCommand.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                tableNameInfo.getTbl(), PrivPredicate.SHOW)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_TABLE_ACCESS_DENIED_ERROR,
+                    PrivPredicate.SHOW.getPrivs().toString(), tableNameInfo.getTbl());
+        }
+    }
+
+    /**
+     * refresh table.
+     */
+    private void refreshTable(String catalogName, String dbName, String tableName, boolean ignoreIfNotExists)
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/refresh/RefreshTableCommand.java:90]:
```suggestion
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/refresh/RefreshTableCommand.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        refreshTableInternal(catalog, db, table, 0);
+
+        ExternalObjectLog log = new ExternalObjectLog();
+        log.setCatalogId(catalog.getId());
+        log.setDbId(db.getId());
+        log.setTableId(table.getId());
+        Env.getCurrentEnv().getEditLog().logRefreshExternalTable(log);
+    }
+
+    private void refreshTableInternal(CatalogIf catalog, DatabaseIf db, TableIf table, long updateTime) {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/refresh/RefreshTableCommand.java:125]:
```suggestion
```

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**